### PR TITLE
Phpstan

### PR DESCRIPTION
**Pull request by escopecz**
[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | N
| New feature? | Y
| Automated tests included? | Y
| Related user documentation PR URL | /
| Related developer documentation PR URL | /
| Issues addressed (#s or URLs) | /
| BC breaks? | N
| Deprecations? | N

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:

It makes sense that after this great refactoring we should use tools that will help us discover issues. PHPSTAN is perfect for that. We couldn't use it on Mautic 2 as it's PHP 7+ only but we can use it now. This PR adds it to the dev dependencies, runs it on Travis CI. It required some changes. I did the simple changes in this PR and I created other issues for those that will take some time.

I also broke running PHPSTAN and CS Fixer into separate Travis CI jobs. Check how Travis looks like now.

This PR must be rebased once those issues are solved:
- [x] https://github.com/mautic/mautic/pull/8270
- [x] https://github.com/mautic/mautic/issues/8297
- [x] https://github.com/mautic/mautic/issues/8294
- [x] https://github.com/mautic/mautic/issues/8293
- [x] https://github.com/mautic/mautic/issues/8292

#### Steps to test this PR:
1. Load up [this PR](https://mautibox.com)
2. Make sure Mautic loads. Click around. No error should appear.

### CODE REVIEW

**Comment by hluchas** [comment]
PR https://github.com/mautic/mautic/pull/8307/files contains code from this branch and it's not readable for review for now.

How about the workflow ?... hmm.

**Comment by hluchas** [comment]
Fixes for #8294 #8293 #8292 #8297 here https://github.com/mautic/mautic/pull/8311

File: app/bundles/EmailBundle/MonitoredEmail/Mailbox.php [hunk 1]
```diff
@@ -281,21 +281,14 @@ public function setImapPath($settings = null)
...
-        /* @var int $port */
-        /* @var string $encryption */
-        /* @var string $folder */
-        /* @var string $user */
-        /* @var string $password */
-        extract($settings);
-        if (!isset($encryption)) {
-            $encryption = (!empty($ssl)) ? '/ssl' : '';
+        if (!isset($settings['encryption'])) {
+            $settings['encryption'] = (!empty($ssl)) ? '/ssl' : '';
```
[the remark this answers is not in the document]
Comment (escopecz) [reply app/bundles/EmailBundle/MonitoredEmail/Mailbox.php]: Fixed in https://github.com/mautic/mautic/pull/8295/commits/3084c766b0d6cfe5044119a26009b44d0003fc21

File: app/bundles/PageBundle/Model/PageModel.php [hunk 2]
```diff
@@ -965,8 +965,7 @@ public function getDeviceGranularityData(\DateTime $dateFrom, \DateTime $dateTo,
         $q->groupBy('ds.device');
 
         $results = $q->execute()->fetchAll();
-
-        $chart = new PieChart($data['labels']);
```
[the remark this answers is not in the document]
Comment (escopecz) [reply app/bundles/PageBundle/Model/PageModel.php]: Fixed in https://github.com/mautic/mautic/pull/8295/commits/17dd1fdc9ed06077484fd480903a98ee59252421

File: plugins/MauticCrmBundle/Integration/SugarcrmIntegration.php [hunk 3]
```diff
@@ -51,15 +49,13 @@ class SugarcrmIntegration extends CrmAbstractIntegration
 
     private $sugarDncKeys = ['email_opt_out', 'invalid_email'];
     private $authorzationError;
```
[the remark this answers is not in the document]
Comment (escopecz) [reply plugins/MauticCrmBundle/Integration/SugarcrmIntegration.php]: Fixed in https://github.com/mautic/mautic/pull/8295/commits/653b06df78a1350586195d4119496359aced0bd7

File: app/bundles/PageBundle/Model/TrackableModel.php [hunk 4]
```diff
@@ -879,7 +874,6 @@ private function parseContent($content, $channel, $channelId, array &$trackableT
                 // HTML5
                 'mautic:disable-tracking'    => '',
             ],
-            'first_pass'  => [],
```
[the remark this answers is not in the document]
Comment (escopecz) [reply app/bundles/PageBundle/Model/TrackableModel.php]: Reverted in https://github.com/mautic/mautic/pull/8295/commits/a933fb533b631d0148f14a6bab62ca0e7adcfab0

**Pull request merged**